### PR TITLE
692 search available tenants

### PR DESCRIPTION
**Pull request by jhosep-marin**
Search Available Tenants (#692)

- Adds Search Tenant gRPC endpoint.
- Enables gRPC endpoint in lhctl.
- Updates Proto schema with respective definitions used in new endpoint.
- Adds exception that gets thrown whenever a principal tries to assign tenant resources as perTenantACLs.
- Adds test methods that provide Tenant ACLs.

### CODE REVIEW

File: server/src/main/java/io/littlehorse/common/model/getable/objectId/TenantIdModel.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public String toString() {
 
     @Override
     public void initFromString(String storeKey) {
-        this.id = storeKey;
+        this.id = storeKey.split("/")[1];
```
Comment (coltmcnealy-lh) [inline server/src/main/java/io/littlehorse/common/model/getable/objectId/TenantIdModel.java:47]: Why is this necessary? This looks like it could permit many bugs wherein I pass in a `WfSpecId` and get out a `TenantId` for example.

File: server/src/main/java/io/littlehorse/server/KafkaStreamsServerImpl.java [hunk 2]
```diff
@@ -39,166 +29,24 @@
...
-import io.littlehorse.sdk.common.proto.ListExternalEventsRequest;
-import io.littlehorse.sdk.common.proto.ListNodeRunsRequest;
-import io.littlehorse.sdk.common.proto.ListTaskMetricsRequest;
-import io.littlehorse.sdk.common.proto.ListTaskMetricsResponse;
-import io.littlehorse.sdk.common.proto.ListTaskRunsRequest;
-import io.littlehorse.sdk.common.proto.ListUserTaskRunRequest;
-import io.littlehorse.sdk.common.proto.ListVariablesRequest;
-import io.littlehorse.sdk.common.proto.ListWfMetricsRequest;
-import io.littlehorse.sdk.common.proto.ListWfMetricsResponse;
+import io.littlehorse.sdk.common.proto.*;
```
Comment (coltmcnealy-lh) [inline server/src/main/java/io/littlehorse/server/KafkaStreamsServerImpl.java:39]: Hmm, @eduwercamacaro do we have a style guide for these changes?

**Review by coltmcnealy-lh — COMMENTED** [review]
I will need to test some of the other search functionality here...this PR seems much more difficult than I thought it would be.

As a rule, we are going to have non-implementers do testing of PR's before they get merged

File: server/src/main/java/io/littlehorse/common/model/getable/objectId/TenantIdModel.java (hunk 1 above)
Comment (eduwercamacaro) [reply server/src/main/java/io/littlehorse/common/model/getable/objectId/TenantIdModel.java:47]: When we are iterating the result of a range scan, we pass the full store key to the `ObjectIdModel#fromString` method. This is very common with the other ObjectId objects see `TaskRunIdModel.initFromString` , and I believe we should do the same for this `TenantIdModel`. 
Could you please explain to me how this can allow many bugs that are not present in the previous version of this class?

File: server/src/main/java/io/littlehorse/server/KafkaStreamsServerImpl.java (hunk 2 above)
Comment (eduwercamacaro) [reply server/src/main/java/io/littlehorse/server/KafkaStreamsServerImpl.java:39]: spotless only verify linting problems. In order to add a validation for this can of changes we need a tool like Checkstyle. see ([AvoidStarImport](https://checkstyle.sourceforge.io/checks/imports/avoidstarimport.html#AvoidStarImport))

Comment (coltmcnealy-lh) [reply server/src/main/java/io/littlehorse/server/KafkaStreamsServerImpl.java:39]: Do you have an opinion about these changes?

**Comment by coltmcnealy-lh** [comment]
TODO's:

1. Prevent having a `/` in a `TenantId` in order to prevent injection attacks. For example, the following code causes a crash:
```
lhctl put tenant 123/123
lhctl search tenant
```
2. Rename `objectIdExtractorPattern` -> `tenantScopedObjectIdExtractorPattern`
3. `TenantIdModel#initFromString` should accept a `objectId`. It should not do a split()

File: server/src/main/java/io/littlehorse/server/KafkaStreamsServerImpl.java (hunk 2 above)
Comment (eduwercamacaro) [reply server/src/main/java/io/littlehorse/server/KafkaStreamsServerImpl.java:39]: we should not use wildcard imports

**Comment by jhosep-marin** [comment]
@coltmcnealy-lh I think the latest requested changes are in place. Please, have a look one more time, and let me know if there is anything else we need to do here.

**Review by coltmcnealy-lh — APPROVED** [review]
